src/lib/operators/projection.cpp

```diff
@@ -81,6 +82,40 @@ std::shared_ptr<const Table> Projection::_on_execute() {
         output_segments[column_id] = input_chunk->get_segment(pqp_column_expression->column_id);
         column_is_nullable[column_id] =
             column_is_nullable[column_id] || input_table.column_is_nullable(pqp_column_expression->column_id);
+      } else if (expression->type == ExpressionType::PQPColumn && !forward_columns) {
+        // Cannot immediately forward the column because it is a ReferenceSegment and we need to return a ValueSegment.
```

Comment:
> Cannot immediately forward the column because it is a ReferenceSegment and we need to return a ValueSegment.

I did not get this straight away. Add a comment about how you came to the above conclusion. E.g.:

```cpp
// In this branch, one column gets projected without any modifications. For other columns, however, 
// the projection does more than that. Therefore, we have to return a ValueSegment. Since forward_columns 
// is set to false, the input segment must be of type ReferenceSegment and needs to be materialized. (...)
```

Reply:
Mmh, I can't point to it, but I found that one hard to follow. I tried to merge the two:

> The current column will be returned without any logical modifications. As other columns do get modified (and returned as a ValueSegment), all segments (including this one) need to become ValueSegments. This segment is not yet a ValueSegment (otherwise forward_columns would be true); thus we need to materialize it.

Better?

Reply:
Very good! :)